content/architecture/index.textile

```diff
@@ -0,0 +1,141 @@
+* **Fault tolerance**: Faults in the system are expected, and the system must have redundancy at every layer in the stack to ensure availability and reliability.
+* **Autonomy**: Each component in the system should be able to operate fully without reliance on a global controller. For example, two isolated data centers should continue to service realtime requests while isolated.
+* **Consistent low latencies**: Within data centers, we aim for latencies to be in the low 10s of milliseconds and less than 100ms globally. Consistently achieving low latencies requires careful consideration of the placement of data and services across the system as well as prioritisation of the computation performed by each service.
+* **Quality of service**: We intentionally design for specific integrity and consistency properties to enable sophisticated realtime applications that would be impossible on platforms with weaker guarantees.
+
+h3. Cluster Architecture
+
+Ably's platform runs on AWS EC2 infrastructure with a globally distributed architecture. Our "clusters":/docs/platform-customization#dedicated-and-isolated-clusters typically span multiple regions, usually between two and ten. This multi-region approach maximizes availability and is a critical aspect of providing a fault-tolerant service.
+
+Each regional deployment operates independently, handling its own subscriber connections, REST traffic, channel management and message routing. When activity occurs on a channel across multiple regions, messages flow peer-to-peer between regions directly, eliminating central bottlenecks and single points of failure.
+
+Our architecture consists of four primary layers:
+
+* **Routing Layer**: Provides intelligent, latency optimised routing for robust end client connectivity.
+* **Frontend Layer**: Handles REST requests and maintains realtime connections (such as WebSocket, Comet and SSE).
+* **Core Layer**: Performs all central message processing for channels.
+* **Gossip Layer**: Distributes network topology information and facilitates service discovery.
+
+Each component scales independently in each region based on demand. We continuously monitor CPU, memory, and other key metrics, triggering autoscaling based on aggregated performance indicators.
+
+The key to our horizontal scalability is intelligent load distribution that efficiently utilizes new capacity as it becomes available:
+
+* In the frontend layer, new instances join the load balancer pool and begin handling their share of incoming connections and REST requests.
+* In the core layer, we employ consistent hashing to distribute channels across core processes - each core maintains a set of pseudorandomly-generated hashes that determine channel placement. As the cluster scales, channels automatically relocate to maintain even load distribution.
+
+h3. Routing Layer
```

Comment:
I think all the layers are correct to be h3, when cluster architecture is correctly at h2.

Reply:
Fixed in: [59617e3](https://github.com/ably/docs/pull/2496/commits/59617e315db0bd03de9ff940fe6dcc45928d470d)